src/journal/ObjectRecorder.cc

```diff
@@ -59,10 +69,20 @@ bool ObjectRecorder::append(const AppendBuffers &append_buffers) {
         last_flushed_future = iter->first;
       }
     }
+
+    if (unlock) {
```

Comment:
Delay the unlock so that you don't need to (potentially) immediately re-lock/unlock the lock below.